### PR TITLE
fix incorrect count of comments

### LINKED ISSUE #1045: New comments do not update Proposal Comment counter
The image shows "all 11 comments" on the comments list, and this number gets increased as comments are added. However, the "8 comments" (below "ABANDON" button) doesn't.

When you refresh the page, the number is changed.

Captura de Tela 2019-03-18 às 19 55 07

### PR DESCRIPTION
**Pull request by camus-code**
Closes #1045

### CODE REVIEW

File: src/selectors/app.js [hunk 1]
```diff
@@ -104,6 +104,7 @@ export const getEditProposalValues = state => {
 
   const files = name ? getNotMarkdownFile(state) : [];
   const description = name ? getTextFromIndexMd(getMarkdownFile(state)) : "";
+  console.log(name, description);
```
Comment (victorgcramos) [inline src/selectors/app.js]: You left a log here!

Comment (camus-code) [reply src/selectors/app.js]: thanks for the catch! it's removed now

File: src/connectors/thingLink.js [hunk 2]
```diff
@@ -1,9 +1,11 @@
+import React from "react";
 import { connect } from "react-redux";
 import * as sel from "../selectors";
 import * as act from "../actions";
 import get from "lodash/fp/get";
 import compose from "lodash/fp/compose";
 import { arg } from "../lib/fp";
+import { buildCommentsTree, buildSetOfComments } from "../lib/snew";
```
Comment (fernandoabolafio) [inline src/connectors/thingLink.js]: Why do you need to build the comments tree just to get the number of comments? Wouldn't it be just as simple as getting the length of the proposal comments? See my comments for ThingLink.js.

File: src/components/snew/ThingLink.js [hunk 3]
```diff
@@ -86,6 +86,7 @@ class ThingLinkComp extends React.Component {
       otherFiles,
       review_status,
       numcomments,
+      numofcomments,
```
Comment (fernandoabolafio) [inline src/components/snew/ThingLink.js]:
```suggestion

```

File: src/components/snew/ThingLink.js [hunk 4]
```diff
@@ -188,6 +189,12 @@ class ThingLinkComp extends React.Component {
       }
     };
 
+    // Fetches inital comment count from api to prevent flickering
+    const correctNumComments =
```
Comment (fernandoabolafio) [inline src/components/snew/ThingLink.js]:
```suggestion
    const correctNumComments = comments.length;
```

File: src/lib/api.js [hunk 5]
```diff
@@ -334,14 +334,14 @@ export const proposalSetStatus = (email, csrf, token, status, censorMsg) =>
...
+  POST("/proposals/new", csrf, proposal)
+    .then(({ response: { censorshiprecord } }) => ({
       ...proposal,
       censorshiprecord,
       timestamp: Date.now() / 1000,
       status: PROPOSAL_STATUS_UNREVIEWED
-    })
-  );
+    }))
+    .then(console.log({ ...proposal }));
```
Comment (fernandoabolafio) [inline src/lib/api.js]: Remove this

File: src/connectors/thingLink.js (hunk 2 above)
Comment (camus-code) [reply src/connectors/thingLink.js]: Upon closer inspection you're right. I mistakenly believed that `numcomments` only updated upon page refresh, but getting the length of this prop returns the correct result. Addressing this now

Comment (fernandoabolafio) [reply src/connectors/thingLink.js]: Just so you do not get confused, `numcomments` is a field returned with the proposal data which is indeed updated upon page refresh. The prop `comments` which I am suggesting you to use to check the length is updated every time a new comment is received.

Comment (camus-code) [reply src/connectors/thingLink.js]: Ahh okay I appreciate the clarification. I had forgotten we also had the prop `comments` and assumed the tree had to be built to get the length.

Updated PR is up :)

File: src/selectors/app.js (hunk 1 above)
Comment (tiagoalvesdulce) [inline src/selectors/app.js]: You left a log here

**Comment by camus-code** [comment]
@fernandoabolafio rebased :)

**Pull request merged**